src/Microsoft.AspNet.Mvc.Core/ModelBinding/MutableObjectModelBinder.cs

```diff
@@ -109,6 +115,13 @@ protected virtual bool CanUpdateProperty([NotNull] ModelMetadata propertyMetadat
 
         private async Task<bool> CanValueBindAnyModelProperties(MutableObjectBinderContext context)
         {
+            // If there are no properties on the model, there is nothing to bind. We are here means this is not a top 
+            // level object. So we return false.
+            if (context.PropertyMetadata == null || context.PropertyMetadata.Count == 0)
```

Comment:
.Any() instead of .Count seems cheaper and allows PropertyMetadata to stay IEnumerable<ModelMetadata>.